rs/ic_os/os_tools/hostos_tool/src/main.rs

```diff
@@ -56,48 +56,46 @@ pub fn main() -> Result<()> {
 
     match opts.command {
         Some(Commands::GenerateNetworkConfig { output_directory }) => {
-            let config_map = config_map_from_path(Path::new(&opts.config))
-                .context("Please specify a valid config file with '--config'")?;
+            let config_map = config_map_from_path(Path::new(&opts.config)).context(format!(
+                "Failed to get config.ini settings for path: {}",
+                &opts.config
+            ))?;
             eprintln!("Using config: {:?}", config_map);
 
             let network_info = NetworkInfo::from_config_map(&config_map)?;
             eprintln!("Network info config: {:?}", &network_info);
 
-            let deployment_settings = get_deployment_settings(Path::new(&opts.deployment_file));
-
-            let deployment_name: Option<&str> = match &deployment_settings {
-                Ok(deployment) => Some(deployment.deployment.name.as_str()),
-                Err(e) => {
-                    eprintln!("Error retrieving deployment file: {e}. Continuing without it");
-                    None
-                }
-            };
-
-            let mgmt_mac: Option<&str> = match &deployment_settings {
-                Ok(deployment) => deployment.deployment.mgmt_mac.as_deref(),
-                Err(_) => None,
-            };
-
-            generate_network_config(
-                &network_info,
-                mgmt_mac,
-                deployment_name,
-                NodeType::HostOS,
-                Path::new(&output_directory),
-            )
-        }
-        Some(Commands::GenerateIpv6Address { node_type }) => {
             let deployment_settings = get_deployment_settings(Path::new(&opts.deployment_file))
-                .context("Please specify a valid deployment file with '--deployment-file'")?;
+                .context(format!(
+                    "Failed to get deployment settings for file: {}",
+                    &opts.deployment_file
+                ))?;
             eprintln!("Deployment config: {:?}", deployment_settings);
 
-            let config_map = config_map_from_path(Path::new(&opts.config))
-                .context("Please specify a valid config file with '--config'")?;
+            let deployment_name = deployment_settings.deployment.name.as_str();
+            let mgmt_mac = deployment_settings.deployment.mgmt_mac.as_deref();
+            let generated_mac = generate_mac_address(deployment_name, &NodeType::HostOS, mgmt_mac)?;
```

Comment:
Awesome! What do you think about also lifting the branch in `generate_mac_address` to this level, too?

Right now, that function also handles if a dev/test `mgmt_mac` is passed, so it doesn't always "generate".

Reply:
In the top-level, this would look something like:

```
let generated_mac = if let Some(mgmt_mac) = deployment_settings.deployment.mgmt_mac.as_deref() {
    let mgmt_mac = FormattedMacAddress::try_from(mgmt_mac)?;
    eprintln!(
        "Using mgmt_mac address found in deployment.json: {}",
        mgmt_mac.get()
    );
    Ok(mgmt_mac)
} else {
    generate_mac_address(
        deployment_settings.deployment.name.as_str(),
        &NodeType::HostOS,
        None,
    )?
};
```

My only concern is this would add a lot of repeat code (three times in hostos_tool and twice in setupos_tool), and then there's a big conditional in all the top level commands rather than it being contained to generate_mac_address.

Perhaps instead we can just rename generate_mac_address?
- resolve_mac_address
- get_or_generate_mac_address
- fetch_mac_address

Reply:
Ak sorry, I got this turned around since the last time we looked at it. 😣 I have this wrong:

> Right now, that function also handles if a dev/test mgmt_mac is passed, so it doesn't always "generate".

We _do_ always generate a new mac, and `mgmt_mac` feeds into this generation. If it is `None`, we call out to `ipmitool`.

So, even with this PR, we're passing `mgmt_mac` around as optional. If we bring that to the top it would look something like:
```
let mgmt_mac = match deployment_settings.deployment.mgmt_mac {
    Some(config_mac) => config_mac,
    None => get_ipmi_mac(),
};
let generated_mac = generate_mac_address(
    deployment_settings.deployment.name.as_str(),
    &NodeType::HostOS,
    mgmt_mac,
)?;
...
generate_network_config(&network_info, generated_mac, Path::new(&output_directory))
```

(After this, maybe it is fine that `generate_mac_address` stays inside `generate_network_config`, but it might be good to still clean up the optional `deployment_name`?)

Reply:
Ah, I see.

To see how it would look, I moved the mgmt_mac match to the top level to see how it would look. WDYT now? Options

- leave like this
- move mgmt_mac conditional back into generate_mac_address (the way this PR was before this recent change)
- move the mgmt_mac conditional and call to generate_mac_address back into generate_network_config (the way the code originally was, and go back to passing around the mgmt_mac optional)
- leave the mgmt_mac conditional in the top level but move the (new) call to generate_mac_address back into generate_network_config

Reply:
Awesome! I think it's a lot more clear to see `mgmt_mac` resolved early, and it should help us with any testing split, so to me the duplication is worth it.

I can see both ways about `generate_mac_address`, whatever you think.

Reply:
I think I'll leave it as is then! That way we don't have to pass the deployment name and node type to generate_network_config

Reply:
Actually I don't know. Maybe it would now be cleaner to move the call to generate_mac_address back into generate_network_config. I can't decide 😮‍💨

Reply:
Okay, I think I like it the way it is because then there's more consistency with the other commands (GenerateIpv6Address, GenerateMacAddress). They all call generate_mac_address directly.

Reply:
Sounds good to me!